ui/src/admin/containers/chronograf/UsersPage.js

```diff
@@ -0,0 +1,126 @@
+import React, {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import * as adminChronografActionCreators from 'src/admin/actions/chronograf'
+import {publishAutoDismissingNotification} from 'shared/dispatchers'
+
+import EmptyUsersTable from 'src/admin/components/chronograf/EmptyUsersTable'
+import UsersTable from 'src/admin/components/chronograf/UsersTable'
+
+class UsersPage extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isLoading: true,
+    }
+  }
+
+  handleCreateUser = user => {
+    const {links, actions: {createUserAsync}} = this.props
+    createUserAsync(links.users, user)
+  }
+
+  handleUpdateUserRole = (user, currentRole, {name}) => {
+    const {actions: {updateUserAsync}} = this.props
+    const updatedRole = {...currentRole, name}
+    const newRoles = user.roles.map(
+      r => (r.organization === currentRole.organization ? updatedRole : r)
+    )
+    updateUserAsync(user, {...user, roles: newRoles})
+  }
+
+  handleUpdateUserSuperAdmin = (user, superAdmin) => {
+    const {actions: {updateUserAsync}} = this.props
+    const updatedUser = {...user, superAdmin}
+    updateUserAsync(user, updatedUser)
+  }
+
+  handleDeleteUser = user => {
+    const {actions: {deleteUserAsync}} = this.props
+    deleteUserAsync(user)
+  }
+
+  async componentWillMount() {
+    const {
+      links,
+      actions: {loadOrganizationsAsync, loadUsersAsync},
+    } = this.props
+
+    this.setState({isLoading: true})
+
+    loadOrganizationsAsync(links.organizations)
+    await loadUsersAsync(links.users)
+
+    this.setState({isLoading: false})
+  }
+
+  render() {
+    const {
+      meCurrentOrganization,
+      organizations,
+      meID,
+      users,
+      notify,
+    } = this.props
+    const {isLoading} = this.state
+
+    if (isLoading || !(organizations.length && users.length)) {
```

Comment:
This is a little confusing. We have two boolean expressions that are basically checking for the same thing, except each is doing it partially, and so both are needed to fully ensure that the page has all the data it needs.

We should probably be using `Promise.all()` to wait on both `loadOrganizationsAsync` and `loadUsersAsync` in `componentWillMount`. This will allow them to execute in parallel, but to ensure that both are complete before setting `isLoading` to `false`.

Then on line 69, we can simply ask `if (isLoading)`.

Reply:
Sounds good -- thanks